Asgard/Communications/Classes/CbusCanFrameProcessor.cs

```diff
@@ -35,16 +35,33 @@ public string ConstructTransportString(ICbusCanFrame frame)
             return ts.ToString();
         }
 
-        public CbusCanFrame ParseFrame(string transportString)
+        public CbusCanFrame? ParseFrame(string transportString)
         {
             this.logger?.LogTrace("Parsing frame from transport string: {0}", transportString);
             var p = 1;
-            if (transportString[p] != 'S')
+            var canFrameType = transportString[p] switch
             {
-                //non-standard, don't support yet
-                throw new NotImplementedException("Only standard frames are supported.");
+                'S' => CanFrameTypes.Standard,
+                'X' => CanFrameTypes.Extended,
+                _ => CanFrameTypes.Undefined,
+            };
+
+            if (canFrameType == CanFrameTypes.Standard)
+            {
+                return ParseFrameStandard(p, transportString);
+            }
+
+            if (canFrameType == CanFrameTypes.Extended)
+            {
+                return ParseFrameExtended(p, transportString);
             }
 
+            // Ignore all non-Standard and non-Extended frames.
+            return null;
```

Comment:
Either this level or the place calling this should probably the log fact we've not parsed it successfully

Reply:
I think there's a lack of a logger, so the caller would need to handle that. Throwing an exception might well be the best approach: let the exception ripple up to somewhere that can handle the logging.